src/views/statusView.ts

```diff
@@ -157,7 +157,28 @@ export default class StatusView implements vscode.Disposable {
 
     public languageServiceUpdating(fileUri: string): void {
         let bar = this.getStatusBar(fileUri);
-        bar.statusLanguageService.text =  Constants.updatingIntelliSenseLabel;
+        bar.statusLanguageService.text = Constants.updatingIntelliSenseLabel;
+        this.showStatusBarItem(fileUri, bar.statusLanguageService);
+    }
+
+    public getlanguageServiceStatus(fileUri: string): string {
+        let bar = this.getStatusBar(fileUri);
+        return bar.statusLanguageService.text;
+    }
+
+    public languageServiceStatusChanged(fileUri: string, status: string): void {
```

Comment:
I think having both `languageServiceUpdating` and `languageServiceStatusChanged` is a little redundant. They could probably be merged into `languageServiceStatusChanged`, adding one more case to the `switch`.